src/main/java/javaslang/jackson/datatype/deserialize/EitherDeserializer.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2015 The Javaslang Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package javaslang.jackson.datatype.deserialize;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JavaType;
+import javaslang.control.Either;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+class EitherDeserializer extends ValueDeserializer<Either<?, ?>> {
+
+    private static final long serialVersionUID = 1L;
+
+    private final JavaType javaType;
+
+    EitherDeserializer(JavaType valueType) {
+        super(valueType, 2);
+        this.javaType = valueType;
+    }
+
+    @Override
+    public Either<?, ?> deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+        List<Object> list = new ArrayList<>();
+        int typeCounter = 0;
+        while (p.nextToken() != JsonToken.END_ARRAY) {
+            if (typeCounter >= deserializersCount()) {
+                throw ctxt.mappingException(javaType.getRawClass());
```

Comment:
Could you add test to verify this branch?
I believe decoding of something like <code>"[1, 2, 3]"</code> will be enough